### ISSUE TITLE
Compiler issues on Type Resolution while executing `detektTest` on `2.0.0-alpha.3`

### ISSUE THREAD

**Issue by P-py** [detekt/detekt#9297]
## Expected Behavior
Updating detekt from version `1.23.8` to `2.0.0-alpha.3` and for all the verifications to work after necessary configuration changes.

## Observed Behavior
After version update, as said before from `1.23.8` to `2.0.0-alpha.3`, the execution of the task `detektTest` generated by the gradle plugin integration returns multiple random compiler low level errors on type resolution.

The behavior only appears to happen with `detektTest` and not with `detekt` or `detektMain`.

> Before reporting the issue, I tried searching for matches here on the official GitHub repo or anywhere else on Internet but didn't find any solves or similar reports except from [this one](https://github.com/detekt/detekt/issues/9121) - already tried the suggestion in there with `parallel` flag but did not work for my case.

```
4: Task failed with an exception.
-----------
* What went wrong:
Execution failed for task ':...:detektTest'.
> java.lang.IllegalStateException: Analyzing \\wsl.localhost\Ubuntu-22.04\home\sant\...\...\...\src\test\kotlin\com\...\...\...\...\...\...\.kt led to an exception.
  Location: java.base/java.util.concurrent.CompletableFuture.wrapInCompletionException(CompletableFuture.java:325)
  The original exception message was: org.jetbrains.kotlin.utils.exceptions.KotlinIllegalArgumentExceptionWithAttachments: `findFirCompiledSymbol` only works on compiled declarations, but the given declaration is not compiled.
  Running detekt '2.0.0-alpha.3' on Java '24.0.2+12' on OS 'Linux'
  If the exception message does not help, please feel free to create an issue on our GitHub page.

* Try:
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights from a Build Scan (powered by Develocity).
> Get more help at https://help.gradle.org.

* Exception is:
org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':...:detektTest'.
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.lambda$executeIfValid$1(ExecuteActionsTaskExecuter.java:135)
	at org.gradle.internal.Try$Failure.ifSuccessfulOrElse(Try.java:288)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeIfValid(ExecuteActionsTaskExecuter.java:133)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:121)
	at org.gradle.api.internal.tasks.execution.ProblemsTaskPathTrackingTaskExecuter.execute(ProblemsTaskPathTrackingTaskExecuter.java:41)
	at org.gradle.api.internal.tasks.execution.FinalizePropertiesTaskExecuter.execute(FinalizePropertiesTaskExecuter.java:46)
	at org.gradle.api.internal.tasks.execution.ResolveTaskExecutionModeExecuter.execute(ResolveTaskExecutionModeExecuter.java:51)
	at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:57)
	at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:74)
	at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:36)
	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.executeTask(EventFiringTaskExecuter.java:77)
	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:55)
	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:52)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:209)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:204)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:66)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:59)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:166)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:59)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.call(DefaultBuildOperationRunner.java:53)
	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter.execute(EventFiringTaskExecuter.java:52)
	at org.gradle.execution.plan.DefaultNodeExecutor.executeLocalTaskNode(DefaultNodeExecutor.java:55)
	at org.gradle.execution.plan.DefaultNodeExecutor.execute(DefaultNodeExecutor.java:34)
	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:355)
	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:343)
	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.lambda$execute$0(DefaultTaskExecutionGraph.java:339)
	at org.gradle.internal.operations.CurrentBuildOperationRef.with(CurrentBuildOperationRef.java:84)
	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:339)
	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:328)
	at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.execute(DefaultPlanExecutor.java:459)
	at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.run(DefaultPlanExecutor.java:376)
	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
	at org.gradle.internal.concurrent.AbstractManagedExecutor$1.run(AbstractManagedExecutor.java:47)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1095)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:619)
	at java.base/java.lang.Thread.run(Thread.java:1447)
Caused by: org.gradle.api.GradleException: java.lang.IllegalStateException: Analyzing \\wsl.localhost\Ubuntu-22.04\home\sant\...\...\...\src\test\kotlin\...\...\...\...\...\...\...\.kt led to an exception.
Location: java.base/java.util.concurrent.CompletableFuture.wrapInCompletionException(CompletableFuture.java:325)
The original exception message was: org.jetbrains.kotlin.utils.exceptions.KotlinIllegalArgumentExceptionWithAttachments: `findFirCompiledSymbol` only works on compiled declarations, but the given declaration is not compiled.
Running detekt '2.0.0-alpha.3' on Java '24.0.2+12' on OS 'Linux'
If the exception message does not help, please feel free to create an issue on our GitHub page.
	at dev.detekt.gradle.invoke.DetektInvokerKt.processResult(DetektInvoker.kt:88)
	at dev.detekt.gradle.invoke.DetektInvokerKt.access$processResult(DetektInvoker.kt:1)
	at dev.detekt.gradle.invoke.DefaultCliInvoker.invokeCli(DetektInvoker.kt:76)
	at dev.detekt.gradle.Detekt.check(Detekt.kt:234)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
	at java.base/java.lang.reflect.Method.invoke(Method.java:565)
	at org.gradle.internal.reflect.JavaMethod.invoke(JavaMethod.java:125)
	at org.gradle.api.internal.project.taskfactory.StandardTaskAction.doExecute(StandardTaskAction.java:58)
	at org.gradle.api.internal.project.taskfactory.StandardTaskAction.execute(StandardTaskAction.java:51)
	at org.gradle.api.internal.project.taskfactory.StandardTaskAction.execute(StandardTaskAction.java:29)
	at org.gradle.api.internal.tasks.execution.TaskExecution$3.run(TaskExecution.java:259)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$1.execute(DefaultBuildOperationRunner.java:29)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$1.execute(DefaultBuildOperationRunner.java:26)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:66)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:59)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:166)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:59)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.run(DefaultBuildOperationRunner.java:47)
	at org.gradle.api.internal.tasks.execution.TaskExecution.executeAction(TaskExecution.java:244)
	at org.gradle.api.internal.tasks.execution.TaskExecution.executeActions(TaskExecution.java:227)
	at org.gradle.api.internal.tasks.execution.TaskExecution.executeWithPreviousOutputFiles(TaskExecution.java:210)
	at org.gradle.api.internal.tasks.execution.TaskExecution.execute(TaskExecution.java:176)
	at org.gradle.internal.execution.steps.ExecuteStep.executeInternal(ExecuteStep.java:110)
	at org.gradle.internal.execution.steps.ExecuteStep.access$000(ExecuteStep.java:47)
	at org.gradle.internal.execution.steps.ExecuteStep$1.call(ExecuteStep.java:64)
	at org.gradle.internal.execution.steps.ExecuteStep$1.call(ExecuteStep.java:61)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:209)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:204)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:66)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:59)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:166)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:59)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.call(DefaultBuildOperationRunner.java:53)
	at org.gradle.internal.execution.steps.ExecuteStep.execute(ExecuteStep.java:61)
	at org.gradle.internal.execution.steps.ExecuteStep.execute(ExecuteStep.java:47)
	at org.gradle.internal.execution.steps.CancelExecutionStep.execute(CancelExecutionStep.java:42)
	at org.gradle.internal.execution.steps.TimeoutStep.executeWithoutTimeout(TimeoutStep.java:75)
	at org.gradle.internal.execution.steps.TimeoutStep.execute(TimeoutStep.java:55)
	at org.gradle.internal.execution.steps.PreCreateOutputParentsStep.execute(PreCreateOutputParentsStep.java:51)
	at org.gradle.internal.execution.steps.PreCreateOutputParentsStep.execute(PreCreateOutputParentsStep.java:29)
	at org.gradle.internal.execution.steps.RemovePreviousOutputsStep.executeMutable(RemovePreviousOutputsStep.java:70)
	at org.gradle.internal.execution.steps.RemovePreviousOutputsStep.executeMutable(RemovePreviousOutputsStep.java:40)
	at org.gradle.internal.execution.steps.MutableStep.execute(MutableStep.java:25)
	at org.gradle.internal.execution.steps.BroadcastChangingOutputsStep.execute(BroadcastChangingOutputsStep.java:62)
	at org.gradle.internal.execution.steps.BroadcastChangingOutputsStep.execute(BroadcastChangingOutputsStep.java:27)
	at org.gradle.internal.execution.steps.CaptureOutputsAfterExecutionStep.execute(CaptureOutputsAfterExecutionStep.java:69)
	at org.gradle.internal.execution.steps.CaptureOutputsAfterExecutionStep.execute(CaptureOutputsAfterExecutionStep.java:46)
	at org.gradle.internal.execution.steps.ResolveInputChangesStep.executeMutable(ResolveInputChangesStep.java:39)
	at org.gradle.internal.execution.steps.ResolveInputChangesStep.executeMutable(ResolveInputChangesStep.java:28)
	at org.gradle.internal.execution.steps.MutableStep.execute(MutableStep.java:25)
	at org.gradle.internal.execution.steps.BuildCacheStep.executeWithoutCache(BuildCacheStep.java:192)
	at org.gradle.internal.execution.steps.BuildCacheStep.executeAndStoreInCache(BuildCacheStep.java:148)
	at org.gradle.internal.execution.steps.BuildCacheStep.lambda$executeWithCache$4(BuildCacheStep.java:104)
	at java.base/java.util.Optional.orElseGet(Optional.java:364)
	at org.gradle.internal.execution.steps.BuildCacheStep.lambda$executeWithCache$5(BuildCacheStep.java:104)
	at org.gradle.internal.Try$Success.map(Try.java:170)
	at org.gradle.internal.execution.steps.BuildCacheStep.executeWithCache(BuildCacheStep.java:88)
	at org.gradle.internal.execution.steps.BuildCacheStep.lambda$execute$0(BuildCacheStep.java:75)
	at org.gradle.internal.Either$Left.fold(Either.java:116)
	at org.gradle.internal.execution.caching.CachingState.fold(CachingState.java:62)
	at org.gradle.internal.execution.steps.BuildCacheStep.execute(BuildCacheStep.java:74)
	at org.gradle.internal.execution.steps.BuildCacheStep.execute(BuildCacheStep.java:49)
	at org.gradle.internal.execution.steps.StoreExecutionStateStep.execute(StoreExecutionStateStep.java:46)
	at org.gradle.internal.execution.steps.StoreExecutionStateStep.execute(StoreExecutionStateStep.java:35)
	at org.gradle.internal.execution.steps.SkipUpToDateStep.executeBecause(SkipUpToDateStep.java:75)
	at org.gradle.internal.execution.steps.SkipUpToDateStep.lambda$execute$2(SkipUpToDateStep.java:53)
	at java.base/java.util.Optional.orElseGet(Optional.java:364)
	at org.gradle.internal.execution.steps.SkipUpToDateStep.execute(SkipUpToDateStep.java:53)
	at org.gradle.internal.execution.steps.SkipUpToDateStep.execute(SkipUpToDateStep.java:35)
	at org.gradle.internal.execution.steps.legacy.MarkSnapshottingInputsFinishedStep.execute(MarkSnapshottingInputsFinishedStep.java:37)
	at org.gradle.internal.execution.steps.legacy.MarkSnapshottingInputsFinishedStep.execute(MarkSnapshottingInputsFinishedStep.java:27)
	at org.gradle.internal.execution.steps.ResolveIncrementalCachingStateStep.executeDelegate(ResolveIncrementalCachingStateStep.java:49)
	at org.gradle.internal.execution.steps.ResolveIncrementalCachingStateStep.executeDelegate(ResolveIncrementalCachingStateStep.java:27)
	at org.gradle.internal.execution.steps.AbstractResolveCachingStateStep.execute(AbstractResolveCachingStateStep.java:71)
	at org.gradle.internal.execution.steps.AbstractResolveCachingStateStep.execute(AbstractResolveCachingStateStep.java:39)
	at org.gradle.internal.execution.steps.ResolveChangesStep.executeMutable(ResolveChangesStep.java:63)
	at org.gradle.internal.execution.steps.ResolveChangesStep.executeMutable(ResolveChangesStep.java:34)
	at org.gradle.internal.execution.steps.MutableStep.execute(MutableStep.java:25)
	at org.gradle.internal.execution.steps.ValidateStep.execute(ValidateStep.java:64)
	at org.gradle.internal.execution.steps.ValidateStep.execute(ValidateStep.java:42)
	at org.gradle.internal.execution.steps.AbstractCaptureStateBeforeExecutionStep.execute(AbstractCaptureStateBeforeExecutionStep.java:77)
	at org.gradle.internal.execution.steps.AbstractCaptureStateBeforeExecutionStep.execute(AbstractCaptureStateBeforeExecutionStep.java:46)
	at org.gradle.internal.execution.steps.AbstractSkipEmptyWorkStep.executeWithNonEmptySources(AbstractSkipEmptyWorkStep.java:133)
	at org.gradle.internal.execution.steps.AbstractSkipEmptyWorkStep.execute(AbstractSkipEmptyWorkStep.java:64)
	at org.gradle.internal.execution.steps.AbstractSkipEmptyWorkStep.execute(AbstractSkipEmptyWorkStep.java:36)
	at org.gradle.internal.execution.steps.legacy.MarkSnapshottingInputsStartedStep.execute(MarkSnapshottingInputsStartedStep.java:38)
	at org.gradle.internal.execution.steps.LoadPreviousExecutionStateStep.execute(LoadPreviousExecutionStateStep.java:36)
	at org.gradle.internal.execution.steps.LoadPreviousExecutionStateStep.execute(LoadPreviousExecutionStateStep.java:23)
	at org.gradle.internal.execution.steps.HandleStaleOutputsStep.executeMutable(HandleStaleOutputsStep.java:77)
	at org.gradle.internal.execution.steps.HandleStaleOutputsStep.executeMutable(HandleStaleOutputsStep.java:43)
	at org.gradle.internal.execution.steps.MutableStep.execute(MutableStep.java:25)
	at org.gradle.internal.execution.steps.AssignMutableWorkspaceStep.lambda$execute$0(AssignMutableWorkspaceStep.java:35)
	at org.gradle.api.internal.tasks.execution.TaskExecution$4.withWorkspace(TaskExecution.java:305)
	at org.gradle.internal.execution.steps.AssignMutableWorkspaceStep.execute(AssignMutableWorkspaceStep.java:31)
	at org.gradle.internal.execution.steps.AssignMutableWorkspaceStep.execute(AssignMutableWorkspaceStep.java:22)
	at org.gradle.internal.execution.steps.ChoosePipelineStep.execute(ChoosePipelineStep.java:40)
	at org.gradle.internal.execution.steps.ChoosePipelineStep.execute(ChoosePipelineStep.java:23)
	at org.gradle.internal.execution.steps.ExecuteWorkBuildOperationFiringStep.lambda$execute$2(ExecuteWorkBuildOperationFiringStep.java:67)
	at java.base/java.util.Optional.orElseGet(Optional.java:364)
	at org.gradle.internal.execution.steps.ExecuteWorkBuildOperationFiringStep.execute(ExecuteWorkBuildOperationFiringStep.java:67)
	at org.gradle.internal.execution.steps.ExecuteWorkBuildOperationFiringStep.execute(ExecuteWorkBuildOperationFiringStep.java:39)
	at org.gradle.internal.execution.steps.IdentityCacheStep.execute(IdentityCacheStep.java:46)
	at org.gradle.internal.execution.steps.IdentityCacheStep.execute(IdentityCacheStep.java:34)
	at org.gradle.internal.execution.steps.IdentifyStep.execute(IdentifyStep.java:44)
	at org.gradle.internal.execution.steps.IdentifyStep.execute(IdentifyStep.java:31)
	at org.gradle.internal.execution.impl.DefaultExecutionEngine$1.execute(DefaultExecutionEngine.java:68)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeIfValid(ExecuteActionsTaskExecuter.java:132)
	... 33 more
Caused by: java.lang.reflect.InvocationTargetException
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:119)
	at java.base/java.lang.reflect.Method.invoke(Method.java:565)
	at dev.detekt.gradle.invoke.DefaultCliInvoker.invokeCli(DetektInvoker.kt:74)
	... 137 more
Caused by: java.util.concurrent.CompletionException: java.lang.IllegalStateException: Analyzing \\wsl.localhost\Ubuntu-22.04\home\sant\...\...\...\src\test\kotlin\com\...\...\...\...\...\...\.kt led to an exception.
Location: java.base/java.util.concurrent.CompletableFuture.wrapInCompletionException(CompletableFuture.java:325)
The original exception message was: org.jetbrains.kotlin.utils.exceptions.KotlinIllegalArgumentExceptionWithAttachments: `findFirCompiledSymbol` only works on compiled declarations, but the given declaration is not compiled.
Running detekt '2.0.0-alpha.3' on Java '24.0.2+12' on OS 'Linux'
If the exception message does not help, please feel free to create an issue on our GitHub page.
	at java.base/java.util.concurrent.CompletableFuture.wrapInCompletionException(CompletableFuture.java:325)
	at java.base/java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:361)
	at java.base/java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:366)
	at java.base/java.util.concurrent.CompletableFuture.uniExceptionally(CompletableFuture.java:1040)
	at java.base/java.util.concurrent.CompletableFuture$UniExceptionally.tryFire(CompletableFuture.java:1020)
	at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:556)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1819)
	... 3 more
Caused by: java.lang.IllegalStateException: Analyzing \\wsl.localhost\Ubuntu-22.04\home\sant\...\...\...\src\test\kotlin\com\...\...\...\...\...\...\.kt led to an exception.
Location: java.base/java.util.concurrent.CompletableFuture.wrapInCompletionException(CompletableFuture.java:325)
The original exception message was: org.jetbrains.kotlin.utils.exceptions.KotlinIllegalArgumentExceptionWithAttachments: `findFirCompiledSymbol` only works on compiled declarations, but the given declaration is not compiled.
Running detekt '2.0.0-alpha.3' on Java '24.0.2+12' on OS 'Linux'
If the exception message does not help, please feel free to create an issue on our GitHub page.
	at dev.detekt.core.AnalyzerKt.throwIllegalStateException(Analyzer.kt:108)
	at dev.detekt.core.AnalyzerKt.access$throwIllegalStateException(Analyzer.kt:1)
	at dev.detekt.core.Analyzer.runAsync$lambda$0$1(Analyzer.kt:59)
	at dev.detekt.core.TaskPoolKt.recover$lambda$0(TaskPool.kt:13)
	at java.base/java.util.concurrent.CompletableFuture.uniExceptionally(CompletableFuture.java:1036)
	... 6 more
Caused by: java.util.concurrent.CompletionException: org.jetbrains.kotlin.utils.exceptions.KotlinIllegalArgumentExceptionWithAttachments: `findFirCompiledSymbol` only works on compiled declarations, but the given declaration is not compiled.
	at java.base/java.util.concurrent.CompletableFuture.wrapInCompletionException(CompletableFuture.java:325)
	at java.base/java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:361)
	at java.base/java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:366)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1816)
	... 3 more
Caused by: org.jetbrains.kotlin.utils.exceptions.KotlinIllegalArgumentExceptionWithAttachments: `findFirCompiledSymbol` only works on compiled declarations, but the given declaration is not compiled.
	at org.jetbrains.kotlin.analysis.low.level.api.fir.api.LLResolutionFacade.findCompiledFirSymbol(LLResolutionFacade.kt:279)
	at org.jetbrains.kotlin.analysis.low.level.api.fir.api.LLResolutionFacade.resolveToFirSymbol$low_level_api_fir(LLResolutionFacade.kt:135)
	at org.jetbrains.kotlin.analysis.low.level.api.fir.api.LowLevelFirApiFacadeKt.resolveToFirSymbol(LowLevelFirApiFacade.kt:37)
	at org.jetbrains.kotlin.analysis.api.fir.components.KaFirTypeProvider.getFirBySymbols$getFirDeclaration(KaFirTypeProvider.kt:851)
	at org.jetbrains.kotlin.analysis.api.fir.components.KaFirTypeProvider.getFirBySymbols(KaFirTypeProvider.kt:267)
	at org.jetbrains.kotlin.analysis.api.fir.components.KaFirTypeProvider.getType(KaFirTypeProvider.kt:187)
	at org.jetbrains.kotlin.analysis.api.impl.base.KaBaseSession.getType(KaBaseSession.kt)
	at dev.detekt.psi.AnnotationExcluderKt.fqNameOrNull(AnnotationExcluder.kt:70)
	at dev.detekt.psi.AnnotationExcluderKt.access$fqNameOrNull(AnnotationExcluder.kt:1)
	at dev.detekt.psi.AnnotationExcluder.isExcluded(AnnotationExcluder.kt:27)
	at dev.detekt.psi.AnnotationExcluder.shouldExclude(AnnotationExcluder.kt:24)
	at dev.detekt.core.suppressors.AnnotationSuppressorKt.isAnnotatedWith(AnnotationSuppressor.kt:34)
	at dev.detekt.core.suppressors.AnnotationSuppressorKt.isAnnotatedWith(AnnotationSuppressor.kt:37)
	at dev.detekt.core.suppressors.AnnotationSuppressorKt.annotationSuppressorFactory$lambda$1(AnnotationSuppressor.kt:24)
	at dev.detekt.core.AnalyzerKt.filterSuppressedFindings(Analyzer.kt:94)
	at dev.detekt.core.AnalyzerKt.access$filterSuppressedFindings(Analyzer.kt:1)
	at dev.detekt.core.Analyzer.analyze(Analyzer.kt:85)
	at dev.detekt.core.Analyzer.runAsync$lambda$0$0(Analyzer.kt:56)
	at dev.detekt.core.TaskPoolKt.task$lambda$0(TaskPool.kt:11)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1814)
	... 3 more

==============================================================================

BUILD FAILED in 1m 59s
76 actionable tasks: 42 executed, 34 up-to-date
Configuration cache entry stored.
08:58:25: Execution finished 'detektTest --stacktrace'.
```

## Context & Environment

*The project uses a __multi-module__ structure configuring detekt on a separate convention applied per module.*
```
project-source
|---.idea/
|----|--- detekt.xml
|--- buildSrc/
|----|--- build.gradle.kts
|----|--- settings.gradle.kts
|----|--- src/main/kotlin
|----|----|---detekt-conventions.gradle.kts
|--- config/
|----|--- detekt/
|----|----|--- baseline.xml
|----|----|--- config.yml
|--- app1/
|----|--- build.gradle.kts
|--- app2/
|----|--- build.gradle.kts
....
```
```kotlin
// on detekt-conventions.gradle.kts
detekt {
    toolVersion = "2.0.0-alpha.3"
    autoCorrect = true
    source.setFrom("src/main/java", "src/main/kotlin")
    /**
     * https://github.com/detekt/detekt/issues/9121
     */
    parallel = false
    config.setFrom("$rootDir/config/detekt/config.yml")
    buildUponDefaultConfig = true
    allRules = false
    baseline = file("$rootDir/config/detekt/baseline.xml")
    disableDefaultRuleSets = false
    debug = true
    ignoreFailures = false
    ignoredBuildTypes = listOf("release")
    ignoredFlavors = listOf("production")
    ignoredVariants = listOf("productionRelease")
    basePath.set(projectDir)
}
tasks.withType<Detekt>().configureEach {
    jvmTarget = "1.8"
    classpath = files(classpath.filter { it.exists() })
}

tasks.withType<DetektCreateBaselineTask>().configureEach {
    jvmTarget = "1.8"
}
```

* Version of **detekt** used: `2.0.0-alpha.3`
* Version of **Gradle** used: `9.3.0`
```bash
> $ ./gradlew --version                                                                                                     ⬡ 22.21.1 [±main ●]

------------------------------------------------------------
Gradle 9.3.0
------------------------------------------------------------

Build time:    2026-01-16 11:14:22 UTC
Revision:      701205ed2f78811508466c8e1952304c2ea869f5

Kotlin:        2.2.21
Groovy:        4.0.29
Ant:           Apache Ant(TM) version 1.10.15 compiled on August 25 2024
Launcher JVM:  24.0.2 (Eclipse Adoptium 24.0.2+12)
Daemon JVM:    /home/sant/.sdkman/candidates/java/24.0.2-tem (no Daemon JVM specified, using current Java home)
OS:            Linux 5.15.167.4-microsoft-standard-WSL2 amd64
```
* Happens in multiple environments & S.Os (our team tried on WSL, Ubuntu and native Windows)
* After the issue, we are not able to use detektTest step on the newest versions.

**Comment by cortinico** [comment]
Can we get a reproducer of some sorts?

**Comment by P-py** [comment]
@cortinico Hello there! Yes, I could try to create some sort of reproducer for this, but since the said project is a private corporate project, I don't think we could share much more details (I even had to hide some of the stacktrace info before publishing the issue😅 ).

It will be hard, but I will try to create a similar environment on a open repository of my own and share it with you. Will return it here when I have something done.

**Comment by jschneider** [comment]
I have a semi-large mono repo here. Running into the same issue from time to time. 
Reproducing is hard - it seems to be a race condition of some sort.
Wild guess: Maybe related to Object2ObjectOpenHashMap which is not thread safe?

**Comment by jweick** [comment]
I had a similar issue running `detektDebug` on `2.0.0-alpha.5`

```
Execution failed for task ':feature:detektDebug' (registered by plugin 'com.android.internal.library').
> Analyzing ...Screen.kt led to an exception.
  Location: org.jetbrains.kotlin.analysis.low.level.api.fir.api.LLResolutionFacade.findCompiledFirSymbol(LLResolutionFacade.kt:280)
  The original exception message was: `findFirCompiledSymbol` only works on compiled declarations, but the given declaration is not compiled.
  Running detekt '2.0.0-alpha.5' on Java '21.0.11+10-LTS' on OS 'Mac OS X'
  If the exception message does not help, please feel free to create an issue on our GitHub page.
```

I found out, that in our case, the problem was a violated rule, that had an exception configured, that didn't take effect:

```
klint:
  MaximumLineLength:
    active: true
    ignoreAnnotated: ['Preview', 'PreviewOnWhiteBackground']
    excludes: [ '**/test/**', '**/androidTest/**']
    maxLineLength: 120
```

But instead of just pointing out the MaximumLineLength-violation, `detektDebug` returned the error shown above.
For us, the solution was to use the `style.MaxLineLength`-rule instead. Luckily there was an alternative.

**Comment by P-py** [comment]
Hello there @jweick! First of all, **thank you so much for the collaboration on this issue**, and the good news is that this change on `config.yml` fixed the issue for my teams project here! Honestly, this saved our pipelines because we were just dealing with that as a software mistery 😄 

- Updated `detekt-formatting` to `ktlint` (had already done that on the initial version update while the issue was still happening)
- Removed the `ktlint.MaximumLineLength` rule block
- Added the `active: true` and `excludes: ["**/test/**"]` to `style.MaxLineLength` block

Sorry for the lack of feedback on this thread @cortinico, on the last months I couldn't take time to work on anything related to this, but this suggestion fix by @jweick may lead to some hints on the real root problem.

**Comment by andrii-zaliskyi** [comment]
I'm hitting the same crash on `2.0.0-alpha.5`, and I traced it to the `ignoreAnnotated` option in my `detekt.yml` and removing every entry makes the analysis run clean. I had it on five rules across different rule sets (including a ktlint-wrapper one), so it looks like the fault is in the shared `ignoreAnnotated` resolution rather than in any single rule. For now I've replaced it with a custom Gradle script that filters annotated functions out of the report and the baseline, but I'd rather drop that and go back to `ignoreAnnotated` once this is fixed.

**Comment by BraisGabin** [comment]
Interesting. I have a hunch. For this issue to appear you need:
- Use the ktlint ruleset
- Add an `ignoreAnnotated` in one of the ktlint rules
- Run detekt in fullMode

A reproducer would help a lot here to be sure. But the KtFiles that ktlint the ruleset run are a bit different than the ones that the rest of rules run so maybe that's the problem for:

> `findFirCompiledSymbol` only works on compiled declarations, but the given declaration is not compiled.

It also explains why I never had this problem (I don't use the ktlint ruleset).

Do you have autocorrect enabled when this issue happens?
